server/remark-includes.ts

```diff
@@ -66,6 +65,23 @@ const numIncludes = (value: string) => value.match(globalIncludeRegexp).length;
 const isInclude = (node: Code | Text): node is Code | Text =>
   typeof node.value === "string" && includeRegexp.test(node.value);
 
+const handlePartialLink = (node, path: string, mdxPath: string) => {
+  if (node.type === "link") {
+    const href = node.url;
+
+    if (typeof href !== "string" || href[0] === "/" || /^http/.test(href)) {
+      return href;
+    }
+
+    const absStart = "docs/pages";
+    const absMdxPath = dirname(absStart + mdxPath.split(absStart).pop());
+    const absTargetPath = join(dirname(path), href);
+
+    node.url = relative(absMdxPath, absTargetPath);
+  }
+  node.children?.forEach((child) => handlePartialLink(child, path, mdxPath));
+};
```

Comment:
Missing a type for `node`: `(node, path: string, mdxPath: string)`

Reply:
I don't have access to the documentation repository. So I can't make changes to the documentation. I can make a short screen-record if you want

Reply:
@Alqanar I am asking for comments added in the code as well as an explanation in this Pull Request so we know how to test this. You could also add a section to the readme in this repository with further explanation. 

```ts
/*
 (Leave comments that explain why this code is needed / how it works, with some examples of how to setup the links.)
*/
const handlePartialLink = (node, path: string, mdxPath: string) => {
  if (node.type === "link") {
    const href = node.url;

    if (typeof href !== "string" || href[0] === "/" || /^http/.test(href)) {
      return href;
    }

    const absStart = "docs/pages";
    const absMdxPath = dirname(absStart + mdxPath.split(absStart).pop());
    const absTargetPath = join(dirname(path), href);

    node.url = relative(absMdxPath, absTargetPath);
  }
  node.children?.forEach((child) => handlePartialLink(child, path, mdxPath));
};
```

There is also a TypeScript issue I noted above that should be fixed.